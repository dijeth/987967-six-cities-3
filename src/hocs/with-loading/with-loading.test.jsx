import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import withLoading from './with-loading.jsx';
import NameSpace from '../../reducers/name-space.js';

const mockStore = configureStore([]);
const Component = () => <div>Mock</div>;
const ComponentWithLoading = withLoading(Component);

it(`should render "Loading..." when "loading" === 1`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      loading: 1
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <ComponentWithLoading />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`should render Component when loading === 0`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      loading: 0
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <ComponentWithLoading />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
