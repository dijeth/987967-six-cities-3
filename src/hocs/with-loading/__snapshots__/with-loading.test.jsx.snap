// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render "Loading..." when "loading" === 1 1`] = `
<h1>
  Loading...
</h1>
`;

exports[`should render Component when loading === 0 1`] = `
<div>
  Mock
</div>
`;
